qt/src/SourceManager.cc

```diff
@@ -44,6 +54,34 @@ QDataStream& operator>>(QDataStream& ds, Source*&) {
 	return ds;
 }
 
+Source::Source(const QString& _path, const QString& _displayname, bool _isTemp)
+				: path(_path), displayname(_displayname), isTemp(_isTemp) {
+	// Check is input PDF file encrypted
+	if(path.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(path));
+		if(document) {
+			if (document->isLocked()) {
+				bool ok = false;
+				QString text;
+				while(true) {
+					text = QInputDialog::getText(MAIN, _("Enter password for the PDF file"),
+														 _("Password:"), QLineEdit::Password,
+														 "", &ok);
+					if(!ok) {
+						throw std::invalid_argument("Locked PDF: skip file.");
+					}
+					if(!document->unlock(text.toLocal8Bit(), text.toLocal8Bit())) {
```

Comment:
Shouldn't that read `if(document->unlock(...))` (without the `!`)?

Reply:
i checked it - '!' is right here.

Reply:
Odd, shouldn't `unlock` return `true` if unlocking was successful and `false` if not?